### PR TITLE
feat: Save button artwork page

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Changes save button copy on artwork page to be "Watch lot" when artwork is in auction.

#### Demo vid:
https://user-images.githubusercontent.com/21178754/225906040-86864d26-26d2-4ea4-9ccc-5f7cb2a9f5fa.mp4

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Save button artwork page copy change - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by dimatretyak — APPROVED** [review]
LGTM 👍

File: src/app/Scenes/Artwork/Components/ArtworkSaveButton.tsx [hunk 1]
```diff
@@ -47,6 +89,9 @@ export const ArtworkSaveButton: React.FC<ArtworkSaveButtonProps> = ({ artwork })
     })
   }
 
+  const a11yLabel = isOpenSale ? "Watch lot" : isSaved ? "Unsave artwork" : "Save artwork"
```
Comment (dimatretyak) [inline src/app/Scenes/Artwork/Components/ArtworkSaveButton.tsx:92]: **question**: should we display unwatch label?

Comment (gkartalis) [reply src/app/Scenes/Artwork/Components/ArtworkSaveButton.tsx:92]: not exactly 😅 the reason is that the label here needs to be in a way descriptive of what the action is while the copy of the button is like the "saved status" 🤔

Comment (dimatretyak) [reply src/app/Scenes/Artwork/Components/ArtworkSaveButton.tsx:92]: **suggestion**: slightly improve the understandability/readability of the code. what do you think?

```javascript
const getA11yLabel = () => {
  if (isOpenSale) {
    return "Watch lot"
  }

  if (isSaved) {
    return "Unsave artwork"
  }

  return "Save artwork"
}
```

**Review by anandaroop — APPROVED** [review]

**Pull request merged**